io-engine/src/grpc/v1/snapshot.rs

```diff
@@ -0,0 +1,431 @@
+use strum::IntoEnumIterator;
+use tonic::{Request, Response, Status};
+
+#[derive(Debug)]
+#[allow(dead_code)]
+pub struct SnapshotService {
+    name: String,
+    client_context: tokio::sync::Mutex<Option<GrpcClientContext>>,
+}
+
+#[derive(Debug)]
+pub struct ReplicaSnapshotDescriptor {
+    pub snapshot_lvol: Lvol,
+    pub replica_uuid: String,
+    pub replica_size: u64,
+}
+impl ReplicaSnapshotDescriptor {
+    fn new(
+        snapshot_lvol: Lvol,
+        replica_uuid: String,
+        replica_size: u64,
+    ) -> Self {
+        Self {
+            snapshot_lvol,
+            replica_uuid,
+            replica_size,
+        }
+    }
+}
+/// Generate SnapshotInfo for the CreateSnapshot Response.
+impl From<ReplicaSnapshotDescriptor> for SnapshotInfo {
+    fn from(r: ReplicaSnapshotDescriptor) -> Self {
+        let snap_lvol = r.snapshot_lvol;
+        let blob = snap_lvol.bs_iter_first();
+        let mut snapshot_param: SnapshotParams = Default::default();
+        for attr in SnapshotXattrs::iter() {
+            let mut val: *const libc::c_char = std::ptr::null::<libc::c_char>();
+            let mut size: u64 = 0;
+            let attr_id = attr.name().to_string().into_cstring();
+            let curr_attr_val = unsafe {
+                let _r = spdk_blob_get_xattr_value(
+                    blob,
+                    attr_id.as_ptr(),
+                    &mut val as *mut *const c_char as *mut *const c_void,
+                    &mut size as *mut u64,
+                );
+
+                let sl =
+                    std::slice::from_raw_parts(val as *const u8, size as usize);
+                std::str::from_utf8(sl).map_or_else(|error| {
+                    warn!(
+                        snapshot=snap_lvol.name(),
+                        attribute=attr.name(),
+                        ?error,
+                        "Failed to parse snapshot attribute, default to empty string"
+                    );
+                    String::default()
+                },
+                |v| v.to_string())
+            };
+            match attr {
+                SnapshotXattrs::ParentId => {
+                    snapshot_param.set_parent_id(curr_attr_val);
+                }
+                SnapshotXattrs::EntityId => {
+                    snapshot_param.set_entity_id(curr_attr_val);
+                }
+                SnapshotXattrs::TxId => {
+                    snapshot_param.set_txn_id(curr_attr_val);
+                }
+                SnapshotXattrs::SnapshotUuid => {
+                    snapshot_param.set_snapshot_uuid(curr_attr_val);
+                }
+            }
+        }
+        Self {
+            snapshot_uuid: snap_lvol.uuid(),
+            snapshot_name: snap_lvol.name(),
+            snapshot_size: snap_lvol.size(),
+            num_clones: 0, //TODO: Need to implement along with clone
+            timestamp: None, //TODO: Need to update xAttr to track timestamp
```

Comment:
Ah, so this is why I'm getting None in the control-plane :)

Reply:
Yes. Created a new task for it. Will work on this.
https://dcsw-bolt.atlassian.net/browse/GTM-939